packages/salesforcedx-vscode-core/src/modes/demo-mode.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+export function isDemoMode() {
+  return process.env.SFDX_ENV ? process.env.SFDX_ENV === 'DEMO' : false;
```

Comment:
@amphro - I've verified this against the release branch of the SFDX CLI that `DEMO` is what we are checking for.